packages/docusaurus-plugin-content-docs/src/globalData.ts

```diff
@@ -56,7 +70,9 @@ export function toGlobalDataVersion(version: LoadedVersion): GlobalVersion {
     isLast: version.isLast,
     path: version.versionPath,
     mainDocId: version.mainDocId,
-    docs: version.docs.map(toGlobalDataDoc),
+    docs: version.docs
+      .map(toGlobalDataDoc)
+      .concat(version.categoryGeneratedIndices.map(toGlobalDataGeneratedIndex)),
```

Comment:
I don't like this part much 😅 generated index are not docs, why not create an extra separate attribute instead of trying to put these in the same array?

The items do not need to have a weird id=slug in such a case

Reply:
On the contrary, those are different things, and the differences should rather be handled in client-side hooks.

If I use `useDocs("xyz")`, and there is a doc with id=xyz + a generated index with slug=xyz, which one should be returned? 

These edge cases can happen

Reply:
So for one thing, all slugs are prefixed with `/`, so there would never be clashes between doc IDs and slugs

Reply:
I understand that this is not very useful and probably confusing, but it improves uniformity while won't introduce any bugs. Maybe we want to formalize "generated index IDs"?

Reply:
> I understand that this is not very useful and probably confusing, but it improves uniformity while won't introduce any bugs. 

We can keep it for now as it's likely to work fine in most cases, but I'd like to change that later

> Maybe we want to formalize "generated index IDs"?

I don't know if this is needed 🤷‍♂️ a slug/permalink should already be unique, the id is just a convenience for the user to reference in sidebar/navbar etc